index.d.ts

```diff
@@ -1,23 +1,27 @@
 import { FastifyPluginCallback } from 'fastify';
 import { Redis, RedisOptions } from 'ioredis';
 
+export interface FastifyRedisNamespacedInstance {
+  [namespace: string]: Redis;
+}
+
+export type FastifyRedis = FastifyRedisNamespacedInstance & Redis;
+
 declare module 'fastify' {
   interface FastifyInstance {
-    redis: Redis;
+    redis: FastifyRedis;
   }
 }
 
-export type FastifyRedisPlugin = RedisOptions &
+export type FastifyRedisPluginOptions = (RedisOptions &
```

Comment:
Public exported name should not be changed as it may break someone using it.

Reply:
Reverted back to `FastifyRedisPlugin` name